### PR TITLE
Transaction integration test

### PR DESCRIPTION
**Pull request by newhoggy**
# Description

We don't previously have a transaction integration test.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import qualified Data.Map as Map
+import qualified Hedgehog.Extras.Test.Concurrent as H
+import           Testnet.Components.SPO
+import qualified Testnet.Process.Run as H
+import           Testnet.Process.Run
+import qualified Testnet.Property.Utils as H
+import           Testnet.Runtime
+
+hprop_transaction :: Property
+hprop_transaction = H.integrationRetryWorkspace 0 "conway-stake-snapshot" $ \tempAbsBasePath' -> do
```
Comment (james-iohk) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs:43]: Use relevant workspace name e.g. `babbage-transaction`

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    , "--out-file", txbodySignedFp
+    ]
+
+  void $ execCli' execConfig
+    [ "babbage", "transaction", "submit"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-file", txbodySignedFp
+    ]
+
+  H.threadDelay 1_000_000
```
Comment (james-iohk) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs:110]: Given the short slot duration, consider asserting for created txo. Can use e.g. [getTxOutAtAddress](https://github.com/input-output-hk/antaeus/blob/828f950db4bd08062817125ecb984710539fe200/e2e-tests/test/Helpers/Query.hs#L179)

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    , wallets
+    } <- cardanoTestnet options conf
+
+  poolNode1 <- H.headM poolNodes
+
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1
+
+  let sbe = ShelleyBasedEraBabbage
```
Comment (Jimbo4350) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs:70]: Defining these two eras separately could lead to issues if only one is changed. The following would be better:
```
let sbe = ShelleyBasedEraBabbage
    era = toCardanoEra sbe
```

**Review by Jimbo4350 — APPROVED** [review]
LGTM. One minor comment.

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs]: Fixed!

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs (hunk 2 above)
Comment (newhoggy) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs]: Asserted the UTxO was created.

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs (hunk 3 above)
Comment (newhoggy) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs]: Fixed!

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs (hunk 2 above)
Comment (james-iohk) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs]: Thanks. I think use of the cli will be less maintenance going forwards than the api directly :grin:

**Review by james-iohk — APPROVED** [review]

**Pull request merged**